griptape/tasks/base_text_input_task.py

```diff
@@ -83,6 +59,30 @@ def all_rulesets(self) -> list[Ruleset]:
 
         return structure_rulesets + task_rulesets
 
+    @property
+    def input(self) -> TextArtifact:
+        if isinstance(self._input, TextArtifact):
+            return self._input
+        elif isinstance(self._input, Callable):
+            return self._input(self)
+        else:
+            return TextArtifact(J2().render_from_string(self._input, **self.full_context))  # type: ignore
+
+    @input.setter
+    def input(self, value: str | TextArtifact | Callable[[BaseTask], TextArtifact]) -> None:
+        self._input = value
+
+    @property
+    def full_context(self) -> dict[str, Any]:
+        if self.structure:
+            structure_context = self.structure.context(self)
+
+            structure_context.update(self.context)
+
+            return structure_context
+        else:
+            return {}
+
```

Comment:
Any particular reason for this move down?

Reply:
this was briefly migrated to a base class and migrated back. Bubbling this back up to keep the diff clean